nps docs (#2778)
mattermost/docs: source/integrations/net-promoter-score.rst

Net Promoter Score Plugin
=========================

> Mattermost is introducing in-product feedback surveys beginning in v5.12. Feedback is collected in the form of a Net Promoter Score survey, and is used to measure user satisfaction and improve product quality by hearing directly from users. 

Reviewer: ```suggestion
Mattermost is introducing in-product feedback surveys beginning in v5.12. Feedback is collected in the form of a Net Promoter Score survey and is used to measure user satisfaction and improve product quality by hearing directly from users. 
```

.. contents::
  :depth: 2
  :local:
  :backlinks: entry

Administration
--------------
Is the survey enabled by default?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The Net Promoter Score (NPS) plugin and surveys are enabled by default on all servers that have `Error and Diagnostic Reporting <https://docs.mattermost.com/administration/telemetry.html>`_ enabled when upgrading to v5.12 or later. 

How can surveys be disabled?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Removing or disabling the **Net Promoter Score** plugin from **System Console > Plugins (Beta) > Plugin Management** will disable surveys. Alternatively, you can disable surveys from the plugin configuration in **System Console > Plugins (Beta) > Net Promoter Score**. 

Reviewer: We could add a note about why one would disable surveys from the plugin configuration.

> If surveys have been disabled from the plugin configuration, users can still send written feedback by messaging Surveybot.

If the plugin or surveys are disabled, they will remain disabled for subsequent server upgrades.

When is the survey scheduled?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Users will receive surveys 21 days after the server is upgraded to v5.12 or later, assuming the following conditions are true:

Reviewer: Yes, it's 21 days since the last upgrade.

> Users will receive surveys 21 days after the server is upgraded to v5.12 or later, assuming the following conditions are true:

Author: Clarified to make that more obvious in the wording

- NPS plugin is enabled in **System Console > Plugins (Beta) > Plugin Management**
- NPS survey is enabled in the plugin configuration in **System Console > Plugins (Beta) > Net Promoter Score**
- User account is greater than 21 days old
- User has not completed a survey in the last 90 days
- User has not been sent a survey in the last 90 days
- Current server version is greater than the server version of the last NPS survey, excluding dot releases

> The above conditions mean that at maximim frequency a user will receive a survey every 90 days, assuming the server is upgraded within that time period. 

Reviewer 2: ```suggestion
The above conditions mean that at maximum frequency a user will receive a survey every 90 days, assuming the server is upgraded within that time period. 
```

How will I be notified a survey is scheduled?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

System Administrators will receive an email notification and in-product Direct Message from "Surveybot" mentioning the scheduled date the survey will be triggered.

.. image:: ../images/nps-admin.png

Survey Data
-----------

How is the survey received?
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Once the survey is triggered on the server, all users will receive an in-product Direct Messages from "Surveybot" on their next interactation with Mattermost. Surveys are not sent to users who are offline.

Reviewer 2: ```suggestion
Once the survey is triggered on the server, all users will receive an in-product Direct Messages from "Surveybot" on their next interaction with Mattermost. Surveys are not sent to users who are offline.
```

> Once the survey is triggered on the server, all users will receive an in-product Direct Messages from "Surveybot" on their next interactation with Mattermost. Surveys are not sent to users who are offline.

Reviewer: Surveys are sent to users when they log in or refresh the page, regardless of their actual status

> Users can optionally select a 0-10 score on how likely they are to recommend Mattermost and then provide written feedback about their expereince. Selecting a score and providing feedback are optional, and the survey can be ignored without interupting usage of Mattermost.

Reviewer 2: ```suggestion
Users can optionally select a 0-10 score on how likely they are to recommend Mattermost and then provide written feedback about their experience. Selecting a score and providing feedback are optional, and the survey can be ignored without interrupting usage of Mattermost.
```

> Users can optionally select a 0-10 score on how likely they are to recommend Mattermost and then provide written feedback about their expereince. Selecting a score and providing feedback are optional, and the survey can be ignored without interupting usage of Mattermost.

Reviewer: Not at this time

.. image:: ../images/nps-survey.png

What data is collected?
~~~~~~~~~~~~~~~~~~~~~~~
Data is only collected when a user selects a score or provides written feedback in response to survey questions. The following non-personally identifiable data is collected:

- Survey information:

>    - ``score``: 0-10 score submitted by the user

Reviewer: 0/5, but we don't really need to include the internal names for these fields
